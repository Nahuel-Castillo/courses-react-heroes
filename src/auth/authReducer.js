import { types } from "../types";

const initialState = {
    name: '',
    logged: false
}

export const authReducer = ( state = initialState, action ) => {

    switch ( action?.type ) {

        case types.login:
            return {
                name: action.payload,
                logged: true
            }
        
        case types.logout:
            return {
                ...initialState
            }

        default:
            return state;
    }

};