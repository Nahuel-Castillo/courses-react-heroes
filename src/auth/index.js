export * from "./AuthContext";
export * from "./authReducer";

// exports = {
//     ...authReducer,
//     ...AuthContext
// }