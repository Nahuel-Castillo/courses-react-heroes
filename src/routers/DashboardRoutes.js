import React from 'react';
import { Redirect, Route, Switch } from 'react-router-dom';
import { DcScreen, HeroScreen, MarvelScreen, SearchScreen, Navbar } from '../components';

export const DashboardRoutes = () => {

    return (
        <>
            <Navbar />  

            <div className="container mt-4">
                <Switch>
                    <Route exact path="/marvel" component={ MarvelScreen }/>
                    <Route exact path="/hero/:heroId" component={ HeroScreen }/>
                    <Route exact path="/dc" component={ DcScreen }/>
                    <Route exact path="/search" component={ SearchScreen }/>
                    <Redirect to="/marvel" component={ MarvelScreen }/>
                </Switch>
            </div>
        </>
    )
}
