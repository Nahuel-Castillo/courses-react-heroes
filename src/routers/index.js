export * from "./AppRouter";
export * from "./DashboardRoutes";
export * from "./PrivateRoute";
export * from "./PublicRoute";

// exports = {
//     ...AppRouter,
//     ...DashboardRoutes,
//     ...PrivateRoute,
//     ...PublicRoute
// }