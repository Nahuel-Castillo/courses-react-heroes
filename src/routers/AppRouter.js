import React, { useContext } from 'react';

import {
    BrowserRouter as Router,
    Switch
  } from 'react-router-dom';
import { AuthContext } from '../auth';
import { LoginScreen } from '../components';
import { PublicRoute, DashboardRoutes, PrivateRoute } from '../routers';

export const AppRouter = () => {

  const { user } = useContext( AuthContext );

  return (
      <Router>
        <div>

          <Switch>
            {/* If the route is login render <LoginScreen/>*/}
            <PublicRoute isAuthenticated={ user?.logged } exact path="/login" component={ LoginScreen } />

            {/* if the route is others render <DashboardRoutes/>*/}
            <PrivateRoute isAuthenticated={ user?.logged } path="/" component={ DashboardRoutes } />

          </Switch>
        </div>
      </Router>
  );
}
