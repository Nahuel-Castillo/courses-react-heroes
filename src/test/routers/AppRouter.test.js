import { mount } from "enzyme";
import { AuthContext } from "../../auth";
import { AppRouter } from "../../routers";

describe('AppRouter test', () => {

    const context = {
        dispatch: jest.fn(),
        user: {
            logged:false
        }
    }
    
    test('should show login if user is no authenticated', () => {
        
        const wrapper = mount( 
            <AuthContext.Provider value={ context }>
                <AppRouter/>
            </AuthContext.Provider>
        );

        expect( wrapper ).toMatchSnapshot();
    });

    test('should show the component <Marvel/> if is authenticated', () => {
        context.user = { name: 'Nahue', logged: true };

        const wrapper = mount( 
            <AuthContext.Provider value={ context }>
                <AppRouter/>
            </AuthContext.Provider>
        );

        expect( wrapper.find('.navbar').exists() ).toBeTruthy();

    });
});