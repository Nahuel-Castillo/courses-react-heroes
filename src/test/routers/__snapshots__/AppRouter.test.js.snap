// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter test should show login if user is no authenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h2>
       LoginScreen 
    </h2>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
